### PR TITLE
Fix big workspace numbers overflow

### LINKED ISSUE #5078: Big workspace numbers overflow
Ref: https://github.com/i3/i3/pull/3950/commits/83c7aff089a6728b6e522d934d656a8e09463112

### PR DESCRIPTION
**Pull request by ammgws**
Fixes #5078, which also brings us inline with i3 behaviour.﻿

### CODE REVIEW

File: sway/ipc-json.c [hunk 1]
```diff
@@ -384,8 +384,19 @@ static json_object *ipc_json_describe_scratchpad_output(void) {
 
 static void ipc_json_describe_workspace(struct sway_workspace *workspace,
 		json_object *object) {
-	int num = isdigit(workspace->name[0]) ? atoi(workspace->name) : -1;
-
+	int num;
+	if isdigit(workspace->name[0]) {
```
Comment (RedSoxFan) [inline sway/ipc-json.c]: There are missing parentheses around the condition. (gcc should not be accepting this...)

Comment (ammgws) [reply sway/ipc-json.c]: @RedSoxFan added parentheses. Either way it built and ran so not sure how to catch this next time.

Comment (RedSoxFan) [reply sway/ipc-json.c]: Apparently gcc implements isdigit as a macro that starts with a paren, which is why it was accepting it. The FreeBSD CI build (https://builds.sr.ht/~sircmpwn/job/167521), which uses clang, did fail and showed the error. Anyway, thanks for fixing it.

**Comment by RedSoxFan** [comment]
Thanks!

**Pull request merged**